examples/dlint/js.rs

```diff
@@ -0,0 +1,249 @@
+        },
+      ),
+    );
+
+    self.runtime.register_op(
+      "add_code",
+      deno_core::json_op_sync(
+        move |state: &mut OpState,
+              args: Value,
+              _bufs: &mut [ZeroCopyBuf]|
+              -> Result<Value, AnyError> {
+          let code_from_js: Code = serde_json::from_value(args).unwrap();
+          let mut stored =
+            state.try_take::<Codes>().unwrap_or_else(HashSet::new);
+          // TODO(magurotuna): To differenciate builtin and plugin rules, adds prefix to plugins.
+          // This should be discussed further before it's stabilized.
+          stored.insert(format!("@deno-lint-plugin/{}", code_from_js.code));
+          state.put::<Codes>(stored);
+
+          Ok(serde_json::json!({}))
+        },
+      ),
+    );
+  }
+}
+
+// TODO(magurotuna): FsModuleLoader is copied from:
+// https://github.com/denoland/deno/pull/8381/files#diff-f7e2ff9248fdb8e71463e0858bfa7070680a09d9704db54d678bf86e49fce3e4
+// This feature is going to be added to `deno_core`, then we should delegate to it.
+struct FsModuleLoader;
+
+impl ModuleLoader for FsModuleLoader {
+  fn resolve(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    specifier: &str,
+    referrer: &str,
+    _is_main: bool,
+  ) -> Result<ModuleSpecifier, AnyError> {
+    Ok(ModuleSpecifier::resolve_import(specifier, referrer)?)
+  }
+
+  fn load(
+    &self,
+    _op_state: Rc<RefCell<OpState>>,
+    module_specifier: &ModuleSpecifier,
+    _maybe_referrer: Option<ModuleSpecifier>,
+    _is_dynamic: bool,
+  ) -> Pin<Box<deno_core::ModuleSourceFuture>> {
+    let module_specifier = module_specifier.clone();
+    async move {
+      let path = module_specifier.as_url().to_file_path().unwrap();
+      let content = std::fs::read_to_string(path)?;
+      let module = deno_core::ModuleSource {
+        code: content,
+        module_url_specified: module_specifier.to_string(),
+        module_url_found: module_specifier.to_string(),
+      };
+      Ok(module)
+    }
+    .boxed_local()
+  }
+}
+
+fn create_dummy_source(plugin_paths: &[&str]) -> String {
+  let mut dummy_source = String::new();
+  for (i, p) in plugin_paths.iter().enumerate() {
+    dummy_source += &format!(
+      "import Plugin{number} from '{path}';\n",
+      number = i,
+      path = p
+    );
+  }
+  dummy_source += r#"Deno.core.ops();
+const programAst = Deno.core.jsonOpSync('get_program', {});
+let plugin;
+"#;
+  for plugin_number in 0..plugin_paths.len() {
+    dummy_source +=
+      &format!("plugin = new Plugin{number}();\n", number = plugin_number);
+    dummy_source += "Deno.core.jsonOpSync('add_diagnostics', plugin.collectDiagnostics(programAst));\n";
+    dummy_source +=
+      "Deno.core.jsonOpSync('add_code', { code: plugin.ruleCode() });\n";
+  }
+
+  dummy_source
+}
+
+impl Plugins for JsRuleRunner {
+  fn run(&mut self, context: &mut Context, program: Program) {
+    self.runtime.register_op(
+      "get_program",
```

Comment:
I know I added this API, but it's debatable, maybe instead we could call global function like `runPlugins()` that would take JSON serialized AST as an argument?